dask/sizeof.py

```diff
@@ -200,3 +210,25 @@ def sizeof_pyarrow_chunked_array(data):
         @sizeof.register(pa.Column)
         def sizeof_pyarrow_column(col):
             return int(_get_col_size(col)) + 1000
+
+
+def _register_entry_point_plugins():
+    """Register sizeof implementations exposed by the entry_point mechanism."""
+    entry_points = importlib_metadata.entry_points()
+
+    try:
+        sizeof_entry_points = entry_points["dask.sizeof"]
```

Comment:
It looks like Python 3.10+ / `importlib_metadata >= 3.9.0` has deprecated `dict` access in favor of `select()`. This is what's causing the CI failures here (which are using `importlib_metadata >= 3.9.0`). To accommodate this, let's add a `hasattr(entry_points, "select")` check to determine which method we should use (see https://github.com/intake/filesystem_spec/pull/680 for a place where a similar change was made).

Reply:
@jrbourbeau yes, this was the reason I originally hoped to use a restricted version of `importlib_metadata`. I agree that your suggestion is going to be the most robust / least invasive going forward :)